R/set_rgcca.R

```diff
@@ -1,141 +1,80 @@
-# inds : individuals removed from the blocks (for crossvalidation)
-# affects same parameters as rgcca_res (or other ones if specified) on a subset 
-# of individuals determined by inds (individuals to remove)
-set_rgcca <- function(
-    rgcca_res,
-    blocks = NULL,
-    connection = NULL,
-    tau = 1,
-    sparsity = 1,
-    ncomp = NULL,
-    scheme = NULL,
-    init = NULL,
-    bias = TRUE,
-    tol = 1e-03,
-    method = NULL,
-    scale = NULL,
-    scale_block = NULL,
-    superblock = NULL,
-    response = NULL,
-    NA_method = NULL,
-    boot = FALSE,
-    inds = NULL) {
-    if(is.null(connection)){    connection <- rgcca_res$call$connection }
-    if(is.null(scale)){    scale <- rgcca_res$call$scale }
-    if(is.null(scale_block)){    scale_block <- rgcca_res$call$scale_block }
-    if(is.null(superblock)){     superblock <- rgcca_res$call$superblock }
-    if(is.null(NA_method)){    NA_method <- rgcca_res$call$NA_method }
-    if(is.null(scheme)){     scheme <- rgcca_res$call$scheme}
-    if(is.null(bias)){     bias <- rgcca_res$call$bias}
-    if(is.null(method)){   method <- rgcca_res$call$method}
-    if(is.null(init)){     init <- rgcca_res$call$init}
-    if(is.null(ncomp)){        ncomp <- rgcca_res$call$ncomp}
-     if (is.null(blocks)) {
-      #  blocks <- rgcca_res$call$blocks
-         blocks = rgcca_res$call$raw
-         blocks = descale(blocks)
-        if (superblock) {
-            for (i in c("tau", "sparsity", "ncomp")) {
-                if (class(rgcca_res$call[[i]]) %in% c("matrix", "data.frame"))
-                  rgcca_res$call[[i]] <- rgcca_res$call[[i]]
-                else
-                  rgcca_res$call[[i]] <- rgcca_res$call[[i]]
-            }
-
-            connection <- NULL
-        }
-
-        sparsity <- rgcca_res$call$sparsity
-        tau <- rgcca_res$call$tau
-
-
-        if (!is.null(rgcca_res$call$response))
-            response <- rgcca_res$call$response
-
-  }else
-#        blocks <- scaling(blocks, scale, scale_block = scale_block)
-
-    if (!boot)
-        blocks <- intersection_list(blocks)
-
-    if (tolower(method) %in% c("sgcca", "spca", "spls")) {
-
-        if (!is.null(blocks) && !missing(tau) && missing(sparsity))
-            stop_rgcca(paste0("sparsity parameter required for ", tolower(method), "instead of tau."))
-
-        par <- "sparsity"
-        penalty <- sparsity
-
-    } else {
-
-        if (!is.null(blocks) && !missing(sparsity) && missing(tau))
-            stop_rgcca(paste0("tau parameter required for ", tolower(method), "instead of sparsity."))
-
-        par <- "tau"
-        penalty <- tau
-    }
-

```

Comment:
This check on `tolower(method)` is already performed in `rgcca`. In order for us to be sure we can remove it, I decided to suppress `method` from the input parameters of `set_rgcca`.